winnow/calibration/calibrator.py

```diff
@@ -90,15 +90,35 @@ def save(cls, calibrator: "ProbabilityCalibrator", path: Path) -> None:
             pickle.dump(calibrator.scaler, f)
 
     @classmethod
-    def load(cls, path: Path) -> "ProbabilityCalibrator":
-        """Load the calibrator from a file.
+    def load(
+        cls,
+        pretrained_model_name_or_path: Union[
+            Path, str
+        ] = "InstaDeepAI/winnow-general-model",
+        cache_dir: Optional[Path] = None,
+    ) -> "ProbabilityCalibrator":
+        """Load a pretrained calibrator from a local path or HuggingFace repository. If the path is a local path, it will be used directly. If it is a HuggingFace repository identifier, it will be downloaded from HuggingFace.
 
         Args:
-            path (Path): The path to load the calibrator from.
-
-        Returns:
-            ProbabilityCalibrator: A new instance of the calibrator loaded from the file.
+            pretrained_model_name_or_path (Union[Path, str]): The local directory path (e.g., Path("./my-model-directory")) or the HuggingFace repository identifier (e.g., "InstaDeepAI/winnow-general-model").
+            cache_dir (Optional[Path]): Directory to cache the HuggingFace model.
         """
+        # Local path if a Path object is provided
+        if isinstance(pretrained_model_name_or_path, Path):
+            path = pretrained_model_name_or_path
+        # HuggingFace repository identifier if a string is provided
+        elif isinstance(pretrained_model_name_or_path, str):
```

Comment:
I feel this is a brittle approach to expect the user to remember to wrap the local path in `Path()` so that you can be able to distinguish between a local path and a HuggingFace identifier.

A more Pythonic approach is to accept both str and Path transparently and then auto-detect whether it’s a local path or a Hugging Face model ID.

```python
path = Path(pretrained_model_name_or_path)

if path.exists():
    # Local path
    resolved_path = path.resolve()
else:
    # Hugging Face repo
    if cache_dir is None: ...
 ```